internal/pkg/api/handleEnroll.go

```diff
@@ -146,6 +154,30 @@ func (et *EnrollerT) _enroll(
 	}
 
 	agentID := u.String()
+	// only delete existing agent if it never checked in
```

Comment:
What do we do if the client gives an enrollmentID for an agent that has checked in?
Would it result in two agents with different IDs that share the same enrollmentID?
Can you add the behaviour to the description of the attribute in the openapi spec?

Reply:
Yes, for an agent that has checked in, there would be 2 agents with the same enrollment_id.

Reply:
Under what circumstances do we think this could happen? A user executing `elastic-agent enroll` on an already enrolled agent?

Reply:
Agent currently doesn't use the `enrollment_id`. This is a safety measure against someone making a destructive action by deleting a healthy agent with an API call with an existing `enrollment_id`